README.md

```diff
@@ -20,38 +20,104 @@ Install it from npm and include it in your React build process (using [Webpack](
 npm install --save react-dropzone
 ```
 
-## Usage
-
-Import `Dropzone` in your React component:
+## Usage with ES2015 classes
 
 ```javascript
+import React from 'react'
+import cx from 'classnames'
 import Dropzone from 'react-dropzone'
-``` 
-  
-  and specify the `onDrop` method that accepts two arguments. The first argument represents the accepted files and the second argument the rejected files.
-  
-```javascript
-function onDrop(acceptedFiles, rejectedFiles) {
-  // do stuff with files...
+
+class MyDropzone extends React.Component {
+
+  onDrop = (acceptedFiles, rejectedFiles) => {
+    // Do something with files
+    // i.e. pre-process, upload etc.
+  }
+
+  render() {
+    return (
+      <Dropzone onDrop={this.onDrop}>
+        {({ getRootProps, getInputProps, isDragActive }) => {
+          return (
+            <div
+              {...getRootProps()}
+              className={cx('dropzone', {
+                'dropzone--isActive': isDragActive
+              })}
+            >
+              <input {...getInputProps()} />
+              {
+                isDragActive ?
+                  <p>Drop files here...</p> :
+                  <p>Try dropping some files here, or click to select files to upload.</p>
+              }
+            </div>
+          )
+        }}
+      </Dropzone>
+    );
+  }
 }
+
 ``` 
 
-Files accepted or rejected based on `accept` prop. This must be a valid [MIME type](http://www.iana.org/assignments/media-types/media-types.xhtml) according to [input element specification](https://www.w3.org/wiki/HTML/Elements/input/file).
+## Render Prop Function
+
+This is where you render whatever you want to based on the state of `Dropzone`.
+You can also pass it as the children prop if you prefer to do things that way
+> `<Dropzone>{/* right here*/}</Dropzone>`
+
+The properties of this object can be split into two categories as indicated
+below:
+
+### Prop Getters
+
+These functions are used to apply props to the elements that you render. This
+gives you maximum flexibility to render what, when, and wherever you like. You
+call these on the element in question (for example: `<div {...getRootProps()}`)). It's advisable to pass all your props to that function
+rather than applying them on the element yourself to avoid your props being
+overridden (or overriding the props returned). For example:
+`getRootProps({onClick(event) {console.log(event)}})`.
+
+| property          | type           | description                                                                    |
+| ----------------- | -------------- | ------------------------------------------------------------------------------ |
+| `getRootProps`    | `function({})` | Returns the props you should apply to the root drop container you render       |
+| `getInputProps`   | `function({})` | Returns the props you should apply to hidden file input you render             |
```

Comment:
I think we might be able to generate the tables with markdown-magic. Nicer to maintain.

Reply:
Ideally these should be generated from code. If that's not possible then yes, let's use some tool. maintaining this will be madness!